src/sentry/auth/access.py

```diff
@@ -103,13 +106,21 @@ class Access(abc.ABC):
     def role(self) -> Optional[str]:
         return self.member.role if self.member else None
 
+    @cached_property
+    def _team_memberships(self) -> Mapping[Team, OrganizationMemberTeam]:
+        if self.member is None:
+            return {}
+        return {
+            omt.team: omt
+            for omt in OrganizationMemberTeam.objects.filter(
+                organizationmember=self.member, is_active=True, team__status=TeamStatus.VISIBLE
+            )
```

Comment:
Use `.select_related("team")` here so that we don't make an extra query for every team